crates/wash-lib/src/build/provider.rs

```diff
@@ -0,0 +1,132 @@
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use std::process;
+
+use anyhow::{anyhow, bail, Context, Result};
+use nkeys::KeyPairType;
+use tracing::warn;
+
+use crate::build::SignConfig;
+use crate::cli::par::{create_provider_archive, detect_arch, ParCreateArgs};
+use crate::cli::{extract_keypair, OutputKind};
+use crate::parser::{CommonConfig, LanguageConfig, ProviderConfig};
+
+/// Build a capability provider for the current machine's architecture
+/// and operating system using provided configuration.
+pub(crate) async fn build_provider(
+    provider_config: &ProviderConfig,
+    language_config: &LanguageConfig,
+    common_config: &CommonConfig,
+    signing_config: Option<&SignConfig>,
+) -> Result<PathBuf> {
+    let LanguageConfig::Rust(rust_config) = language_config else {
+        bail!("Unsupported language for provider: {:?}", language_config)
+    };
+    let mut command = match rust_config.cargo_path.as_ref() {
+        Some(path) => process::Command::new(path),
+        None => process::Command::new("cargo"),
+    };
+
+    // Change directory into the project directory
+    std::env::set_current_dir(&common_config.path)?;
+
+    // Build for a specified target if provided, or the default rust target
+    let build_args = if let Some(override_target) = &provider_config.target {
+        vec!["build", "--release", "--target", override_target]
+    } else {
+        vec!["build", "--release"]
```

Comment:
could probably factor this out and make it mutable to add target if `override_target` is present

Reply:
```
    let mut build_args = Vec::with_capacity(4);
    build_args.extend_from_slice(&["build", "--release"]);
    if let Some(override_target) = &provider_config.target {
        build_args.extend_from_slice(&["--target", override_target]);
    };
```

I think it looks a little weirder but I don't mind, will be better if we need more build args down the line